### PR TITLE
Support Apple silicon natively

### PR DESCRIPTION
**Pull request by lptr**
Fixes https://github.com/gradle/gradle/issues/15509

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QF please.

**Comment by lptr** [comment]
@bot-gradle test this please.

**Comment by lptr** [comment]
@bot-gradle test this please.

**Comment by lptr** [comment]
@bot-gradle test this please.

File: subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public static synchronized void initialize(File userHomeDir, boolean initializeA
                     try {
                         net.rubygrapefruit.platform.Native.init(nativeBaseDir);
                     } catch (NativeIntegrationUnavailableException ex) {
-                        LOGGER.debug("Native-platform is not available.");
+                        LOGGER.debug("Native-platform is not available.", ex);
```
Comment (wolfs) [inline subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java:104]: Does the stacktrace add some valueable information for end-users here?

**Review by wolfs — COMMENTED** [review]
LGTM! I left one comment.

File: subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java (hunk 1 above)
Comment (lptr) [reply subprojects/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java:104]: We log the exception for the `file-events` loading already, and it was needed for me to figure out what was wrong. The cause explained that the library was found but it was of the wrong architecture.

In general I think it's a strong code smell if you catch an exception, log something, but not log the exception.

**Pull request merged**